CodeSnippetsReflection.OData/LanguageGenerators/CommonGenerator.cs

```diff
@@ -360,16 +349,19 @@ public static string LowerCaseFirstLetter(string s)
         public static IEnumerable<KeyValuePair<string, string>> GetParameterListFromOperationSegmentWithNames(OperationSegment operationSegment, SnippetModel snippetModel, bool returnEnumTypeIfEnum, string collectionSuffix = "")
         {
             var parametersProvided = new List<string>();
-            if (!string.IsNullOrEmpty(snippetModel.RequestBody)
-                && JsonConvert.DeserializeObject(snippetModel.RequestBody) is JObject testObj)
+            if (!string.IsNullOrEmpty(snippetModel.RequestBody))
             {
-                foreach (var (key, _) in testObj)
+                using var jsonDoc = JsonDocument.Parse(snippetModel.RequestBody, JsonHelper.JsonDocumentOptions);
+                var jsonObject = jsonDoc.RootElement.Clone();
+                if (jsonDoc.RootElement.ValueKind == JsonValueKind.Object)
```

Comment:
```suggestion
                var jsonObject = JsonSerializer.Deserialize<JsonElement>(snippetModel.RequestBody, JsonHelper.JsonSerializerOptions);
                if (jsonObject.ValueKind == JsonValueKind.Object)
```
Could you change to this instead accross the instances using `JsonDocument.Parse`. Prevents having to create a clone of the element which could be very big.

Reply:
Thanks for the pointer. I've made the update.